### PR TITLE
Add SMB service broker testcase

### PR DESCRIPTION
**Pull request by davewalter**
Thanks for submitting a PR to DRATS.

## Checklist

Please provide the following information:

- [x] What component are you testing?

We are testing the [Samba service broker](https://github.com/cloudfoundry/smb-volume-release).

- [x] Have you created a `TestCase` and added it to the list of cases to be run?

Yes.

- [x] Have you added an `include_<testcase-name>` property and any other new properties to the sample integration_config.json?

Yes

- [x] Have you manually validated your `TestCase` against a deployed Cloud Foundry? If so, which version?

It has been tested against cf-deployment 4.0.0.

- [x] Does this change rely on a particular version of CF release?

No.

- [x] Are you available for a cross-team pair to help troubleshoot your PR?

Yes.

- [x] Have you submitted a pull request to modify the cf-deployment [backup and restore opsfile](https://github.com/cloudfoundry/cf-deployment/blob/master/operations/experimental/enable-backup-restore.yml) to add a backup job and properties where appropriate?

https://github.com/cloudfoundry/cf-deployment/pull/614

### Do you have any other useful information for us?

https://www.pivotaltracker.com/story/show/151415093

I modeled this change on the corresponding NFS volume services testcase that was just merged in PR https://github.com/cloudfoundry-incubator/disaster-recovery-acceptance-tests/pull/40. Let me know if you have any questions.

Regards,
Dave

### CODE REVIEW

**Comment by ChunyiLyu** [comment]
Hi @davewalter,

Thanks for the PR. We are thinking about test and merge the PR after the cf deployment [PR](https://github.com/cloudfoundry/cf-deployment/pull/614) is merged and released on master. Would that be OK? Is this blocking any work in your team?

Cheers,
Chunyi && Mirah

**Comment by davewalter** [comment]
Hi @ChunyiLyu,

That wouldn't block us as we are using the forks used to create both PRs in our CI, but historically, Release Integration has waited to merge their PR until DRATs includes a corresponding test case, so I'm not sure how best to proceed.

Regards,
Dave

**Comment by gmrodgers** [comment]
Hello @davewalter, 

The Release Integration has responded into your PR. They seem to be ok with the current PR to be merged in.  Once this has been done, we can then proceed with merging in this PR to the repo. Thanks!

Glen & @alamages

File: scripts/run_acceptance_tests_with_bbl_env.sh [hunk 1]
```diff
@@ -26,6 +26,19 @@ pushd $1
   else
       echo "Skipping cf-nfsrboker testcase because INCLUDE_NFS_BROKER_TESTCASE is not set to true"
```
Comment (ab-pivot) [inline scripts/run_acceptance_tests_with_bbl_env.sh]: cf-nfsrboker should be cf-nfsbroker

Comment (davewalter) [reply scripts/run_acceptance_tests_with_bbl_env.sh]: Fixed. Thanks for catching this.

**Comment by davewalter** [comment]
Hi @aclevername @gmrodgers,

Both of the PRs I mentioned above have now been merged into cf-deployment and are available in [v5.1.0](https://github.com/cloudfoundry/cf-deployment/releases/tag/v5.1.0).

Please let me know if the new `operations/experimental/use-pxc-for-{nfs,smb}-volume-service.yml` ops-files do not solve the issue you are seeing in CI.

Thanks,
Dave

**Comment by davewalter** [comment]
Hi,

Please hold off on merging this PR for now. We are making changes to the SMB service broker which will require changes here. We will let you know when we are ready to go.

Thanks,
Dave

**Comment by davewalter** [comment]
Hi,

This PR is ready to be pulled in again, but is dependent on the changes in this [PR](https://www.pivotaltracker.com/story/show/161095754) to cf-deployment.

Regards,
Dave

**Comment by aclevername** [comment]
Thanks @davewalter , I've updated our tracker story to block on this PR into cf-deployment

**Comment by ChunyiLyu** [comment]
@davewalter
Hi Dave, 

The PR to cf-deployment was reverted due to a breaking change. We will temporary move this PR into our icebox and pick it up again once the next major version of cf-deployment is released. Is that OK?

Chunyi

**Comment by julian-hj** [comment]
Hey, just a note--this PR should be unblocked, now that cf-d 6 has landed with the latest smb-volume-release ops files in it.

**Comment by gmrodgers** [comment]
Hey squad,

Just as an update.  The bump to v6.x of CF-deployment has broken our pipeline.  This is an issue that the Release Integration team seem to be aware of but until the fix is cut in a release, this PR is blocked.

Thank,
Glen && @aclevername

**Comment by jamesjoshuahill** [comment]
Hi @julian-hj and @davewalter 

We have run this PR again and it failed differently. Details:

- cf-deployment v6.0.0
- ops files:
    - operations/experimental/enable-smb-volume-service.yml
    - operations/backup-and-restore/enable-backup-restore-smb-broker.yml
- errands:
    - smbbrokerpush

DRATS failed during `bbr deployment backup` with this error in the `backup` script:

```
1 error occurred:
error 1:
1 error occurred:
error 1:
Error attempting to run backup for job azurefilebroker-bbr on backup-restore/21dec6f9-b8e1-4edf-bfe7-139462d2aa12:
+ JOB_PATH=/var/vcap/jobs/azurefilebroker-bbr
+ SDK_PATH=/var/vcap/jobs/database-backup-restorer
+ BBR_ARTIFACT_FILE_PATH=/var/vcap/store/bbr-backup/azurefilebroker-bbr//azurefilebroker-db
+ CONFIG_PATH=/var/vcap/jobs/azurefilebroker-bbr/config/config.json
+ /var/vcap/jobs/database-backup-restorer/bin/backup --config /var/vcap/jobs/azurefilebroker-bbr/config/config.json --artifact-file /var/vcap/store/bbr-backup/azurefilebroker-bbr//azurefilebroker-db
ERROR 1045 (28000): Access denied for user 'azurefile-broker'@'10.0.16.6' (using password: YES)
2018/11/13 14:45:19 ERROR 1045 (28000): Access denied for user 'azurefile-broker'@'10.0.16.6' (using password: YES) - exit code 1
```

We ssh'd onto the `backup-restore` VM and tried to connect to the internal MySQL using the `azurefile-broker` username and password defined in `/var/vcap/jobs/azurefilebroker-bbr/config/config.json` and it failed:

```
$ ./mysql -h sql-db.service.cf.internal -u azurefile-broker -p
Enter password:
ERROR 1045 (28000): Access denied for user 'azurefile-broker'@'10.0.16.6' (using password: YES)
```

We confirmed that there is no user called `azurefile-broker` in the `mysql.user` table. How is this user supposed to be created?

Josh and @aclevername

**Comment by davewalter** [comment]
Hi @jamesjoshuahill and @aclevername 

Thanks for taking another look at this. The problem is that the smb-volume-release no longer needs a separate BBR script as it now only uses Credhub for it's backing store, which is backed up by the Credhub BBR script. We have submitted a [PR to cf-deployment](https://github.com/cloudfoundry/cf-deployment/pull/665) to deprecate the ops-file to add the `azurefilebroker-bbr` job to the deployment as it definitely won't work as intended.

Sorry for the continued churn on this one. Hopefully once that PR is merged and a new release available, this will get everything working for you. Let us know if you have any further questions in the meantime.

Regards,
@mariash and Dave

**Comment by julian-hj** [comment]
hey, FWIW, I don't think you need to wait for the new release to move forward on this PR--you just need to make sure to leave the `enable-backup-restore-smb-broker.yml` ops file out of your deployment and you should be good to go.

**Comment by julian-hj** [comment]
excellent thanks!  BTW I noticed that there was a comment in tracker about disabling this test for p-drats.  Since I cannot comment there, I will mention here that you should be able to run this test in p-drats also.  You just have to enable SMB services in your PAS configuration (on the Advanced tab).

Cheers,
-Julian

**Comment by glestaris** [comment]
Thanks, @julian-h. I have relayed your message in the tracker story and tagged the anchors.

Re not having comment access to our tracker: your @dell.com email was already a member and I added your @emc.com and @pivotal.io accounts too. Hopefully one of them will work :)

**Pull request merged**